src/Disks/StoragePolicy.cpp

```diff
@@ -211,7 +211,11 @@ UInt64 StoragePolicy::getMaxUnreservedFreeSpace() const
 {
     UInt64 res = 0;
     for (const auto & volume : volumes)
-        res = std::max(res, volume->getMaxUnreservedFreeSpace());
+    {
+        auto max_unreserved_for_volume = volume->getMaxUnreservedFreeSpace();
+        if (max_unreserved_for_volume)
+            res = std::max(res, *max_unreserved_for_volume);
```

Comment:
`res` will be zero instead of unlimited for the case of all nullopts. Maybe also rework StoragePolicy interface?